### ISSUE TITLE
Essential diagnostics

### ISSUE THREAD

**Issue by dougiesquire** [ACCESS-NRI/access-om3-configs#374]
The set of diagnostics output by `access-om3` configurations is still mostly just inherited from CESM. There are possibly many diagnostics that we want that aren't requested, and others that are requested that we don't want. It would be helpful to compile a set of "essential" diagnostics that should always be requested in configurations so that key metrics/analyses can be calculated.

I'm probably being naive in thinking that it's possible to come up with a single list of essential diagnsotics. In that case, we still need to know what diagnostics are needed to calculate particular metrics so that we can ensure everything we need is saved in upcoming test runs.

**Comment by aidanheerdegen** [comment]
Given we now have our very own Data Master (@joshuatorrance), if we're making sh*t up we should write it down: make our own CV as an extension of the CMIP6 one.

**Comment by aidanheerdegen** [comment]
> ADDED: actually, scratch that. There are the `*Pt` variables. I guess we should be using those...

So ditch `snap`?

**Comment by dougiesquire** [comment]
It was never our intention here to comply with CMIP vocab. Though I agree that having a format that is _sometimes_ consistent with CMIP vocab doesn't help anyone. Currently, we use CMIP vocab _where it's easy_, i.e. where it is easily compatible with `make_diag_table.py`.

But yes, we could use the `*Pt` vocab for snapshots in OM3 - I'll open an issue

**Comment by aidanheerdegen** [comment]
If we don't comply with CMIP we should comply with something* someone** writes down somewhere***


\* Our own CV
** Not me
*** In a repo

**Comment by dougiesquire** [comment]
Yup agreed. It hadn't really occurred to me that we weren't complying until you pointed it out, so I was under the impression that this issue served to document our naming convention (albeit not very transparently). I've just opened [an issue](https://github.com/ACCESS-NRI/access-om3-configs/issues/803) about using the `*Pt` frequencies for OM3 snapshots. If that doesn't go ahead, I'll make sure to write things down more clearly/exhaustively.

**Comment by joshuatorrance** [comment]
This does sound like something that'll be documented in the dataspec soon™.

How do we currently differ from CMIP6's frequency vocab?

**Comment by dougiesquire** [comment]
> How do we currently differ from CMIP6's frequency vocab?

The CMIP vocab are yucky because they encode both frequency time period and reduction method. We do those separately. We use something similar to the CMIP6 CV for the former - basically just using the same abbreviations for month, year etc, with a preceding integer to indicate the number. [This schema](https://github.com/ACCESS-NRI/schema/blob/main/au.org.access-nri/model/output/file-metadata/1-0-1.json) gives our frequency CV:

```
"frequency": {
    "type": "string",
    "oneOf": [
        {
            "pattern": "^fx$"
        },
        {
            "pattern": "^subhr$"
        },
        {
            "pattern": "^\\d+hr$"
        },
        {
            "pattern": "^\\d+day$"
        },
        {
            "pattern": "^\\d+mon$"
        },
        {
            "pattern": "^\\d+yr$"
        },
        {
            "pattern": "^\\d+dec$"
        }
    ],
    "description": "The frequency of the variable(s) in the file asset"
}
```

**Comment by aidanheerdegen** [comment]
If this was still open to modification I'd suggest rather than
```
access-om3.mom6.3d.agessc.rho.d2.1day.mean.1900-01.nc
```

I'd go for something like this
```
access-om3.mom6.3d.agessc+rho_d2.1day.mean.1900-01.nc
```
where the variable and it's mapped coordinate are grouped (don't use the same separator as other fields), and the coordinate is a "modifier" of the variable. It makes it much more straightforward to right a regex:

https://regex101.com/r/8kkeAy/1

The exact characters used within the variable and mapping group don't really matter a great deal, just as long as they're not `.` or a character that might be present in a variable name. It might be fine to re-use the same character (e.g. `+`) for simplicity or if there might be more modifiers than given in the example.

**Comment by dougiesquire** [comment]
> I believe it is possible to break that out into its own separate schema and re-use that in the `file-metadata` schema.

Have you got an example of this? I recall looking briefly into this when I set up those schema and gave up for reasons I don't recall.

**Comment by dougiesquire** [comment]
> Nope. I just read somewhere that it was possible, but have not looked into it.

https://json-schema.org/understanding-json-schema/structuring#dollarref. I'll have a play

**Comment by joshuatorrance** [comment]
> > I believe it is possible to break that out into its own separate schema and re-use that in the `file-metadata` schema.
> 
> Have you got an example of this? I recall looking briefly into this when I set up those schema and gave up for reasons I don't recall.

I've got a very rough draft of a json-schema for the data spec [here](https://github.com/ACCESS-NRI/ACCESS_Output_data_specifications/pull/1/files).

I've used multiple files and `$ref`s that work on my local machine. I have a feeling that I'll need to do things differently to get them working in other contexts though.

**Comment by aekiss** [comment]
Can this issue be closed, or are there outstanding things to address?
